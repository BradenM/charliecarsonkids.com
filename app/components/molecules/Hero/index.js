/**
 *
 * Hero
 *
 * Molecule
 */

import Title from 'atoms/Title';
import PropTypes from 'prop-types';
import React from 'react';
import Button from 'react-bootstrap/Button';
import Col from 'react-bootstrap/Col';
import Container from 'react-bootstrap/Container';
import Jumbotron from 'react-bootstrap/Jumbotron';
import Row from 'react-bootstrap/Row';
import styled from 'styled-components';

const StyledJumbotron = styled(({ imageSrc, ...props }) => (
  <Jumbotron {...props} />
))`
  background: url(${props => props.imageSrc}) center center no-repeat #fafafa;
  background-size: contain;
  min-height: 90vh;
`;

const HeroTitle = styled(Title)`
  text-shadow: 2px 2px 4px rgba(150, 150, 150, 1);
`;

export const Hero = ({ imageSrc, text, buttonText, onButtonClick }) => (
  <Container className="p-0">
    <StyledJumbotron className="hero" imageSrc={imageSrc} fluid>
      <div>
        <Row className="h-100">
          <Col xs={6} className="pb-3 my-auto">
            <HeroTitle className="mb-0 font-weight-bold" color="#fefefe">
              {text.toUpperCase()}
            </HeroTitle>
          </Col>
        </Row>
        <Row>
          <Col xs={3} className="py-3">
            {buttonText && (
              <Button onClick={onButtonClick} size="lg" variant="outline-light">
                {buttonText}
              </Button>
            )}
          </Col>
        </Row>
      </div>
    </StyledJumbotron>
  </Container>
);

Hero.defaultProps = {
  buttonText: '',
  onButtonClick: () => {},
  text: '',
};

Hero.propTypes = {
  imageSrc: PropTypes.string.isRequired,
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  buttonText: PropTypes.string,
  onButtonClick: PropTypes.func,
};

export default Hero;
