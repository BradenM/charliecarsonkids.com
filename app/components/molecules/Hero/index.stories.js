/**
 *
 * Stories for Hero
 *
 * Molecule
 */

import { text } from '@storybook/addon-knobs';
import React from 'react';
import { host } from 'storybook-host';
import { ImageURL } from 'utils/getStatic';
import Hero from '.';

export default {
  title: 'Design|Molecules/Hero',
  component: Hero,
  decorators: [
    host({
      title: 'Hero Component',
      align: 'top left',
      width: '100%',
      height: 500,
      border: true,
    }),
  ],
  parameters: {
    centered: { disable: true },
  },
};

export const defaultHero = () => (
  <Hero text="new arrivals" imageSrc={ImageURL('hero')} />
);

export const withDynamicVariables = () => (
  <Hero text={text('Text', 'new arrivals')} imageSrc={ImageURL('hero')} />
);
