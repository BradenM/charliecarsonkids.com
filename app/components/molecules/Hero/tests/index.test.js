/**
 *
 * Tests for Hero
 *
 * Molecules
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import Hero from '../index';

describe('<Hero />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Hero text="text" imageSrc="image" />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Hero text="hi there" imageSrc="image" />);
    expect(firstChild).toMatchSnapshot();
  });
});
