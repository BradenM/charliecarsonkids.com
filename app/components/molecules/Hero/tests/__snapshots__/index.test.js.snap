// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> Should render and match the snapshot 1`] = `
<div
  class="Hero__StyledJumbotron-sc-1ele3ck-0 hhupmH hero jumbotron"
>
  <div
    class="h-100 container"
  >
    <div
      class="row"
    >
      <div
        class="pb-3 my-auto col-6"
      >
        <h1
          class="Title-sc-3abvrr-0 Hero__HeroTitle-sc-1ele3ck-1 bqYlOi mb-0 font-weight-bold display-1 mb-0 font-weight-bold"
          color="#fefefe"
          size="1"
        >
          HI THERE
        </h1>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="py-3 col-3"
      >
        <button
          class="btn btn-outline-light btn-lg"
          type="button"
        />
      </div>
    </div>
  </div>
</div>
`;
